test/components/form-attachment/list.spec.js

```diff
@@ -1037,15 +1037,31 @@ describe('FormAttachmentList', () => {
   });
 
   describe('dataset linking', () => {
-    beforeEach(() => {
+    it('sends a request for datasets if the project has one', () => {
```

Comment:
This is a test of existing behavior, but I thought it was useful to add. The reverse case — that if the project doesn't have a dataset, then it doesn't send a request — is implicitly tested by all the tests above.